include/wavm/ThreadState.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <mutex>
```

Comment:
I've merged what's left from the deleted headers in the rest of this diff into this single file to make it a bit clearer what's going on